sycl/source/detail/stream_impl.cpp

```diff
@@ -59,21 +61,31 @@ size_t stream_impl::get_size() const { return BufferSize_; }
 
 size_t stream_impl::get_max_statement_size() const { return MaxStatementSize_; }
 
-void stream_impl::flush() {
-  // Access the stream buffer on the host. This access guarantees that kernel is
-  // executed and buffer contains streamed data.
-  {
-    auto HostAcc = detail::Scheduler::getInstance()
-                       .StreamBuffersPool.find(this)
-                       ->second.Buf.get_access<cl::sycl::access::mode::read>(
-                           range<1>(BufferSize_), id<1>(OffsetSize));
+void stream_impl::enqueueFlush() {
+  // We don't want stream flushing to be blocking operation that is why submit a
+  // host task to print stream buffer.
+  queue Q;
```

Comment:
Isn't it better to obtain an existing queue instead of allocating another one?
Host-task will be executed on default host queue regardless of what queue it was enqueued to.
Hence, one may pick default host queue from scheduler:
```suggestion
  queue &Q = cl::sycl::detail::Scheduler::getInstance().getDefaultHostQueue();
```